tests/unit/components/sl-date-picker-test.js

```diff
@@ -367,22 +367,45 @@ test( 'updateDateRange() - when date outside endDate range we show placeholder t
     );
 });
 
-test( 'unregisterEvents() - listens to correct event', function( assert ) {
+test( 'changeDate listener is added and removed from the correct namespace', function( assert ) {
     const component = this.subject();
-    const inputElement = this.$( 'input.date-picker' )[ 0 ];
+    const inputElement = this.$( 'input.date-picker' );
+    const jQueryData = Ember.get( Ember.$, '_data' );
+    const eventData = Ember.get( jQueryData( inputElement[0], 'events' ), 'changeDate' );
```

Comment:
Can't this data be gotten off of `events` via a `.( 'data' )` call instead of using a private, undocumented, property?

Rather than use this approach at all it is preferred to spy on the jQuery `.on()` (or mock if necessary, similar to the mocks in https://github.com/softlayer/sl-ember-components/blob/develop/tests/unit/mixins/sl-tooltip-enabled-test.js#L6-L24) so that we can see that calls with the expected data are made rather than relying on the API to remain unchanged in jQuery, etc.  The point being, what are we after?  We want to make sure that `.on()` is called with a name spaced value.  We are not looking to test the the jQuery implementation of how it happens, whether the jQuery library code is working, etc.